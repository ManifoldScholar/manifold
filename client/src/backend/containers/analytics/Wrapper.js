import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import HeadContent from "global/components/HeadContent";

export default function AnalyticsWrapperContainer({ route }) {
  const { t } = useTranslation();

  return (
    <>
      <HeadContent
        title={`${t("titles.analytics")} | ${t("common.admin")}`}
        appendDefaultTitle
      />
      <main id="skip-to-main" tabIndex={-1}>
        <h1 className="screen-reader-text">{t("dashboard.title")}</h1>
        <section>
          <div className="container">{childRoutes(route)}</div>
        </section>
      </main>
    </>
  );
}

AnalyticsWrapperContainer.propTypes = {
  route: PropTypes.object
};
