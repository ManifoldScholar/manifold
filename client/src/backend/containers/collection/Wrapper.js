import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import Layout from "backend/components/layout";
import Dialog from "backend/components/dialog";
import Navigation from "backend/components/navigation";
import Utility from "global/components/utility";
import { entityStoreActions, notificationActions } from "actions";
import { select } from "utils/entityUtils";
import { collectionsAPI, requests } from "api";
import lh from "helpers/linkHandler";
import { childRoutes, RedirectToFirstMatch } from "helpers/router";
import navigation from "helpers/router/navigation";

import Authorize from "hoc/authorize";

const { request, flush } = entityStoreActions;

export class CollectionWrapperContainer extends PureComponent {
  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      collection: select(requests.beCollection, state.entityStore)
    };
  };

  static displayName = "Collection.Wrapper";

  static propTypes = {
    collection: PropTypes.object,
    dispatch: PropTypes.func,
    match: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      confirmation: null
    };
  }

  componentDidMount() {
    this.fetchCollection();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beCollection));
  }

  fetchCollection = () => {
    const call = collectionsAPI.show(this.props.match.params.id);
    const collectionRequest = request(call, requests.beCollection);
    this.props.dispatch(collectionRequest);
  };

  closeDialog() {
    this.setState({ confirmation: null });
  }

  doPreview = event => {
    event.preventDefault();
    const previewUrl = lh.link(
      "frontendResourceCollection",
      this.props.collection.relationships.project.attributes.slug,
      this.props.collection.attributes.slug
    );
    const win = window.open(previewUrl, "_blank");
    win.focus();
  };

  doDestroy = () => {
    const call = collectionsAPI.destroy(this.props.collection.id);
    const options = { removes: this.props.collection };
    const collectionRequest = request(
      call,
      requests.beCollectionDestroy,
      options
    );
    this.props.dispatch(collectionRequest).promise.then(() => {
      this.notifyDestroy();
      this.redirectToProjectCollections();
    });
  };

  redirectToProjectCollections() {
    const projectId = this.props.collection.relationships.project.id;
    const redirectUrl = lh.link("backendProjectResourceCollections", projectId);
    this.props.history.push(redirectUrl);
  }

  notifyDestroy() {
    const notification = {
      level: 0,
      id: `COLLECTION_DESTROYED_${this.props.collection.id}`,
      heading: "The collection has been destroyed.",
      body: `${
        this.props.collection.attributes.title
      } has passed into the endless night.`,
      expiration: 5000
    };
    this.props.dispatch(notificationActions.addNotification(notification));
  }

  handleCollectionDestroy = event => {
    const heading = "Are you sure you want to delete this collection?";
    const message = "This action cannot be undone.";
    new Promise((resolve, reject) => {
      this.setState({
        confirmation: { resolve, reject, heading, message }
      });
    }).then(
      () => {
        this.doDestroy(event);
        this.closeDialog();
      },
      () => {
        this.closeDialog();
      }
    );
  };

  renderUtility() {
    return (
      <div>
        <button onClick={this.doPreview} className="button-bare-primary">
          <i className="manicon manicon-eye-outline" aria-hidden="true" />
          Preview{" "}
        </button>
        <button
          onClick={this.handleCollectionDestroy}
          className="button-bare-primary"
        >
          <i className="manicon manicon-trashcan" aria-hidden="true" />
          Delete
        </button>
      </div>
    );
  }

  renderRoutes() {
    const { collection } = this.props;
    return childRoutes(this.props.route, { childProps: { collection } });
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { collection, match } = this.props;
    /* eslint-enable no-unused-vars */
    if (!collection) return null;
    const skipId = "skip-to-collection-panel";
    const secondaryLinks = navigation.collection(collection);

    return (
      <div>
        <Authorize
          entity={collection}
          failureFatalError={{
            body: "You are not allowed to edit this collection."
          }}
          ability="update"
        >
          <RedirectToFirstMatch
            from={lh.link("backendCollection", collection.id)}
            candidates={secondaryLinks}
          />
          {this.state.confirmation ? (
            <Dialog.Confirm {...this.state.confirmation} />
          ) : null}
          <Navigation.DetailHeader
            type="collection"
            backUrl={lh.link(
              "backendProjectResourceCollections",
              collection.relationships.project.id
            )}
            backLabel={
              collection.relationships.project.attributes.titlePlaintext
            }
            utility={this.renderUtility()}
            title={collection.attributes.title}
            secondaryLinks={secondaryLinks}
          />
          <Layout.BackendPanel
            sidebar={<Navigation.Secondary links={secondaryLinks} panel />}
          >
            <Utility.SkipLink skipId={skipId} />
            <div id={skipId}>{this.renderRoutes()}</div>
          </Layout.BackendPanel>
        </Authorize>
      </div>
    );
  }
}

export default connectAndFetch(CollectionWrapperContainer);