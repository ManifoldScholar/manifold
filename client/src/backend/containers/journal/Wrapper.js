import React, { useCallback } from "react";
import Layout from "backend/components/layout";
import withConfirmation from "hoc/withConfirmation";
import { journalsAPI } from "api";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import navigation from "helpers/router/navigation";
import Authorize from "hoc/Authorize";
import {
  useFetch,
  useApiCallback,
  useNotification,
  useRedirectToFirstMatch
} from "hooks";
import { useTranslation } from "react-i18next";
import HeadContent from "global/components/HeadContent";
import PageHeader from "backend/components/layout/PageHeader";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";

function JournalWrapper({ match, route, history, confirm, location }) {
  const { t } = useTranslation();
  const { data: journal, refresh } = useFetch({
    request: [journalsAPI.show, match.params.id]
  });
  const destroy = useApiCallback(journalsAPI.destroy, { removes: journal });

  const notifyDestroy = useNotification(j => ({
    level: 0,
    id: `JOURNAL_DESTROYED_${j.id}`,
    heading: t("notifications.journal_delete"),
    body: t("notifications.delete_entity_body", {
      title: j?.attributes?.title
    }),
    expiration: 5000
  }));

  const destroyAndRedirect = useCallback(() => {
    const redirect = () => history.push(lh.link("backendJournals"));
    destroy(journal.id).then(
      () => {
        notifyDestroy(journal);
        redirect();
      },
      () => redirect()
    );
  }, [destroy, history, journal, notifyDestroy]);

  const handleJournalDestroy = useCallback(() => {
    const heading = t("modals.delete_journal");
    const message = t("modals.confirm_body");
    confirm(heading, message, destroyAndRedirect);
  }, [destroyAndRedirect, confirm, t]);

  const utility = [
    {
      label: "actions.view",
      route: "frontendJournalDetail",
      slug: journal?.attributes.slug,
      icon: "eyeOpen32"
    },
    {
      label: "actions.delete",
      authorize: "delete",
      icon: "delete32",
      onClick: handleJournalDestroy
    }
  ];

  const renderRoutes = () => {
    return childRoutes(route, {
      childProps: { refresh, journal }
    });
  };

  useRedirectToFirstMatch({
    route: "backendJournal",
    id: journal?.id,
    slug: journal?.attributes.slug,
    candidates: journal ? navigation.journal(journal) : []
  });

  if (!journal) return null;

  const subpage = location.pathname.split("/")[4]?.replace("-", "_");

  const breadcrumbs = [
    {
      to: lh.link("backendJournals"),
      label: t("glossary.journal_title_case_other")
    },
    {
      to: lh.link("backendJournals", journal.id),
      label: journal.attributes.titlePlaintext
    }
  ];

  return (
    <div>
      <Authorize
        entity={journal}
        failureFatalError={{
          detail: t("journals.unauthorized_edit")
        }}
        ability={["read"]}
      >
        {subpage && (
          <HeadContent
            title={`${t(`titles.${subpage}`)} | ${
              journal.attributes.titlePlaintext
            } | ${t("common.admin")}`}
            appendDefaultTitle
          />
        )}
        <RegisterBreadcrumbs breadcrumbs={breadcrumbs ?? []} />
        <PageHeader
          type="journal"
          title={journal.attributes.titleFormatted}
          subtitle={journal.attributes.subtitle}
          actions={utility}
          secondaryLinks={navigation.journal(journal)}
          issues={journal.attributes.issuesNav}
        />
        <Layout.BackendPanel
          sidebar={
            <Layout.SecondaryNav
              links={navigation.journal(journal)}
              panel
              ariaLabel={t("journals.settings")}
            />
          }
        >
          <div>{renderRoutes()}</div>
        </Layout.BackendPanel>
      </Authorize>
    </div>
  );
}

export default withConfirmation(JournalWrapper);
