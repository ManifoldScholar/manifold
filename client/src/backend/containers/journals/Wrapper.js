import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import Authorize from "hoc/Authorize";

export default function JournalsWrapper({ route }) {
  const { t } = useTranslation();
  return (
    <Authorize
      ability="update"
      entity={["journal"]}
      failureFatalError={{
        body: t("journals.unauthorized")
      }}
    >
      <div>
        <main id="skip-to-main" tabIndex={-1} className="backend-detail">
          {childRoutes(route)}
        </main>
      </div>
    </Authorize>
  );
}

JournalsWrapper.propTypes = {
  route: PropTypes.object
};
