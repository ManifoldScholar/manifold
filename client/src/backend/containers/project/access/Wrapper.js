import React, { Component } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import { childRoutes } from "helpers/router";
import lh from "helpers/linkHandler";
import PermissionsContainer from "backend/containers/permission";
import EntitlementsContainer from "backend/containers/entitlements";
import Authorize from "hoc/Authorize";
import Authorization from "helpers/authorization";
import Layout from "backend/components/layout";
import FormContainer from "global/containers/form";
import Form from "global/components/form";
import Hero from "backend/components/hero";
import withSettings from "hoc/withSettings";
import connectAndFetch from "utils/connectAndFetch";

class ProjectAccessWrapper extends Component {
  static mapStateToProps = state => {
    return {
      authentication: state.authentication
    };
  };

  static displayName = "Project.Access.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    route: PropTypes.object,
    history: PropTypes.object,
    match: PropTypes.object,
    authentication: PropTypes.object,
    t: PropTypes.func
  };

  get defaultIsRestricted() {
    return this.props.settings.attributes.general.restrictedAccess === true;
  }

  get defaultIsOpen() {
    return !this.defaultIsRestricted;
  }

  render() {
    const { project, updateProject, t, authentication } = this.props;
    if (!project) return null;

    const authorization = new Authorization();

    const canGrantPermissions = authorization.authorizeAbility({
      authentication,
      entity: project,
      ability: "managePermissions"
    });

    const closeUrl = lh.link("backendProjectAccess", project.id);
    return (
      <>
        <Authorize
          entity={project}
          ability="update"
          failureNotification
          failureRedirect={lh.link("backendProject", project.id)}
        >
          {canGrantPermissions && (
            <Layout.BackendPanel flush>
              <PermissionsContainer.List entity={project} />
            </Layout.BackendPanel>
          )}
          <Layout.BackendPanel flush={!canGrantPermissions}>
            <EntitlementsContainer.List
              entity={project}
              preList={
                <div style={{ marginBottom: 44, marginTop: 22 }}>
                  <Hero.Block
                    title={t("projects.forms.access.title")}
                    description={t("projects.forms.access.description")}
                    initialVisible={this.defaultIsRestricted}
                  >
                    <FormContainer.Form
                      style={{ paddingTop: 24, paddingBottom: 24 }}
                      model={project}
                      name="backend-project-update"
                      update={updateProject}
                      className="form-secondary"
                    >
                      {this.defaultIsOpen && (
                        <Form.Switch
                          label={t("projects.forms.access.restricted_label")}
                          name="attributes[restrictedAccess]"
                          wide
                        />
                      )}
                      {this.defaultIsRestricted && (
                        <Form.Switch
                          label={t("projects.forms.access.open_access_label")}
                          name="attributes[openAccess]"
                          wide
                        />
                      )}
                      <Form.TextInput
                        label={t("projects.forms.access.notice_header_label")}
                        name="attributes[restrictedAccessHeading]"
                        placeholder={t(
                          "projects.forms.access.notice_header_placeholder"
                        )}
                        wide
                      />
                      <Form.TextArea
                        label={t("projects.forms.access.notice_body_label")}
                        name="attributes[restrictedAccessBody]"
                        placeholder={t(
                          "projects.forms.access.notice_body_placeholder"
                        )}
                      />
                      <Form.Save text={t("projects.forms.access.save")} />
                    </FormContainer.Form>
                  </Hero.Block>
                </div>
              }
            />
          </Layout.BackendPanel>
        </Authorize>
        {childRoutes(this.props.route, {
          drawer: true,
          drawerProps: { closeUrl, lockScroll: "always" },
          childProps: {
            entity: project,
            closeUrl,
            history: this.props.history
          }
        })}
      </>
    );
  }
}

export default withTranslation()(
  withSettings(connectAndFetch(ProjectAccessWrapper))
);
