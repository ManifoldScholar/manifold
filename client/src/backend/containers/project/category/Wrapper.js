import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { childRoutes } from "helpers/router";

class ProjectCategoryWrapperContainer extends Component {
  static displayName = "Project.Category.Wrapper";

  static propTypes = {
    project: PropTypes.object,
    history: PropTypes.object,
    route: PropTypes.object
  };

  render() {
    return (
      <div>{childRoutes(this.props.route, { childProps: this.props })}</div>
    );
  }
}

export default withRouter(ProjectCategoryWrapperContainer);
