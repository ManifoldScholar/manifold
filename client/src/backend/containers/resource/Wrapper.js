import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import connectAndFetch from "utils/connectAndFetch";
import Layout from "backend/components/layout";
import { entityStoreActions, notificationActions } from "actions";
import { select } from "utils/entityUtils";
import { resourcesAPI, requests } from "api";
import lh from "helpers/linkHandler";
import { childRoutes, RedirectToFirstMatch } from "helpers/router";
import navigation from "helpers/router/navigation";
import withConfirmation from "hoc/withConfirmation";
import HeadContent from "global/components/HeadContent";
import PageHeader from "backend/components/layout/PageHeader";
import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";
import { getBreadcrumbs } from "./breadcrumbs";

import Authorize from "hoc/Authorize";

const { request, flush } = entityStoreActions;

export class ResourceWrapperContainer extends PureComponent {
  static mapStateToProps = state => {
    return {
      resource: select(requests.beResource, state.entityStore)
    };
  };

  static displayName = "Resource.Wrapper";

  static propTypes = {
    resource: PropTypes.object,
    match: PropTypes.object,
    dispatch: PropTypes.func,
    history: PropTypes.object,
    route: PropTypes.object,
    confirm: PropTypes.func.isRequired,
    t: PropTypes.func
  };

  static defaultProps = {
    confirm: (heading, message, callback) => callback()
  };

  componentDidMount() {
    this.fetchResource();
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beResource));
  }

  fetchResource = () => {
    const call = resourcesAPI.show(this.props.match.params.id);
    const resourceRequest = request(call, requests.beResource);
    this.props.dispatch(resourceRequest);
  };

  doDestroy = () => {
    const call = resourcesAPI.destroy(this.props.resource.id);
    const options = { removes: this.props.resource };
    const resourceRequest = request(call, requests.beResourceDestroy, options);
    this.props.dispatch(resourceRequest).promise.then(() => {
      this.notifyDestroy();
      this.redirectToProjectResources();
    });
  };

  redirectToProjectResources() {
    const projectId = this.props.resource.relationships.project.id;
    const redirectUrl = lh.link("backendProjectResources", projectId);
    this.props.history.push(redirectUrl);
  }

  notifyDestroy() {
    const t = this.props.t;
    const notification = {
      level: 0,
      id: `RESOURCE_DESTROYED_${this.props.resource.id}`,
      heading: t("notifications.resource_delete"),
      body: t("notifications.delete_entity_body", {
        title: this.props.resource.attributes.title
      }),
      expiration: 5000
    };
    this.props.dispatch(notificationActions.addNotification(notification));
  }

  handleResourceDestroy = () => {
    const t = this.props.t;
    const heading = t("modals.delete_resource");
    const message = t("modals.confirm_body");
    this.props.confirm(heading, message, this.doDestroy);
  };

  get utility() {
    const { resource } = this.props;
    const { project } = resource.relationships ?? {};

    return [
      {
        label: "actions.view",
        route: "frontendProjectResource",
        slug: project.attributes.slug,
        resourceSlug: resource.attributes.slug,
        icon: "eyeOpen32"
      },
      {
        label: "actions.delete",
        icon: "delete32",
        authorize: "delete",
        onClick: this.handleResourceDestroy
      }
    ];
  }

  renderRoutes() {
    const { resource } = this.props;
    return childRoutes(this.props.route, { childProps: { resource } });
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { resource, t } = this.props;
    /* eslint-enable no-unused-vars */
    if (!resource) return null;
    const secondaryLinks = navigation.resource(resource);

    const subpage = location.pathname.split("/")[5];

    const belongsToJournalIssue =
      resource.relationships.project.attributes.isJournalIssue;

    const breadcrumbs = getBreadcrumbs(
      resource,
      resource.relationships.project,
      belongsToJournalIssue,
      t
    );

    const parentProps = {
      parentTitle: resource.relationships.project.attributes.titleFormatted,
      parentSubtitle: resource.relationships.project.attributes.subtitle,
      texts: resource.attributes.projectTextsNav,
      parentId: resource.relationships.project.id
    };

    return (
      <div>
        <Authorize
          entity={resource}
          failureFatalError={{
            body: t("resources.unauthorized")
          }}
          ability="update"
        >
          {subpage && (
            <HeadContent
              title={`${t(`titles.${subpage}`)} | ${
                resource.attributes.titlePlaintext
              } | ${t("common.admin")}`}
              appendDefaultTitle
            />
          )}
          <RedirectToFirstMatch
            route="backendResource"
            id={resource.id}
            slug={resource.attributes.slug}
            candidates={secondaryLinks}
          />
          <RegisterBreadcrumbs breadcrumbs={breadcrumbs ?? []} />
          <PageHeader
            type="resource"
            actions={this.utility}
            title={resource.attributes.titleFormatted}
            subtitle={resource.attributes.subtitle}
            secondaryLinks={secondaryLinks}
            icon="BEResourcesBox64"
            {...parentProps}
          />
          <Layout.BackendPanel
            sidebar={
              <Layout.SecondaryNav
                links={secondaryLinks}
                panel
                ariaLabel={t("resources.settings")}
              />
            }
          >
            <div>{this.renderRoutes()}</div>
          </Layout.BackendPanel>
        </Authorize>
      </div>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(ResourceWrapperContainer))
);
