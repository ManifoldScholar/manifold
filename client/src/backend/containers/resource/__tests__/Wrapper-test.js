import React from "react";
import renderer from "react-test-renderer";
import { ResourceWrapperContainer } from "../Wrapper";
import { wrapWithRouter } from "test/helpers/routing";
import { Provider } from "react-redux";
import build from "test/fixtures/build";

describe("Backend Resource Wrapper Container", () => {
  const store = build.store();
  const resource = build.entity.resource("1");
  resource.relationships.project = build.entity.project("1");
  const currentUser = build.entity.user("1");
  store.dispatch({
    type: "UPDATE_CURRENT_USER",
    error: false,
    payload: {
      data: currentUser
    }
  });

  const component = renderer.create(
    wrapWithRouter(
      <Provider store={store}>
        <ResourceWrapperContainer
          resource={resource}
          route={{
            routes: []
          }}
          match={{
            params: {}
          }}
          dispatch={store.dispatch}
        />
      </Provider>
    )
  );

  it("renders correctly", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("doesn't render to null", () => {
    let tree = component.toJSON();
    expect(tree).not.toBe(null);
  });
});
