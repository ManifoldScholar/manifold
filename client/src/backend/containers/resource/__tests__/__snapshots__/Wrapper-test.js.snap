// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backend/containers/resource/Wrapper matches the snapshot when rendered 1`] = `"<header class=\\"css-oqpkis-Header ecdc3152\\"><div class=\\"css-1k4if3i-Content ecdc3151\\"><div class=\\"css-1ex52a3-Row e12ljl8g17\\"><div class=\\"css-afmy3i-TitleWrapper e12ljl8g4\\"><a class=\\"css-if60nh-Title e12ljl8g3\\" href=\\"/backend/projects/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE\\">Rowan Test</a><span class=\\"css-xbanu3-Subtitle e12ljl8g2\\">World's Greatest Dog</span></div></div><div class=\\"css-1ex52a3-Row e12ljl8g17\\"><div class=\\"css-h1tygr-ChildLink e12ljl8g12\\"></div><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-1y80gnz-TextIcon e12ljl8g5 svg-icon--BEResourcesBox64\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 64 64\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M46.3881227,48.4009376 L46.3918753,30.9062606 L31.8300827,22.6143626 L17.737908,30.9020636 L17.7350935,48.3967216 L32.2969005,56.6886279 L46.3881227,48.4009376 Z M15.7349064,49.5592784 L15.738092,29.7579364 L31.8139173,20.3036374 L48.3921247,29.7437393 L48.3878773,49.5450625 L32.3130995,58.9993722 L15.7349064,49.5592784 Z M32.8980963,13.4767032 L30.8981037,13.4712967 L30.9210037,4.99999672 L32.9209963,5.00540319 L32.8980963,13.4767032 Z M23.7377011,15.6007769 L21.9832989,16.5610231 L17.9564988,9.20392305 L19.7109011,8.24367688 L23.7377011,15.6007769 Z M16.8860026,22.0168924 L15.7903974,23.6901076 L8.97569735,19.2279076 L10.0713026,17.5546924 L16.8860026,22.0168924 Z M53.8650681,16.7125216 L55.024332,18.3422784 L48.387432,23.0631784 L47.2281681,21.4334216 L53.8650681,16.7125216 Z M43.8739203,7.77909374 L45.6640797,8.67090619 L41.9245797,16.1773062 L40.1344203,15.2854938 L43.8739203,7.77909374 Z M31.3083,39.1962962 L33.3083,39.1967038 L33.3045,57.8447038 L31.3045,57.8442963 L31.3083,39.1962962 Z M16.2434424,31.1987741 L17.2331576,29.4608259 L32.8031576,38.3275259 L31.8134424,40.0654741 L16.2434424,31.1987741 Z M46.8323598,29.4637188 L47.8462402,31.1876812 L32.7623402,40.0586812 L31.7484598,38.3347188 L46.8323598,29.4637188 Z\\"></path></svg><div class=\\"css-afmy3i-TitleWrapper e12ljl8g4\\"><h1 class=\\"css-ytpeqb-Title e12ljl8g3\\">Image</h1></div></div><div class=\\"css-hdwovy-Row-DropdownRow e1yvm5522\\"><div class=\\"css-14hqmci-Positioner e2613d75\\"><button tabindex=\\"0\\" type=\\"button\\" aria-expanded=\\"false\\" aria-controls=\\"abcdef\\" class=\\"css-smivd9-Button e2613d77\\"><span>actions.actions</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-klehzp-DropdownIcon e2613d76 svg-icon--disclosureDown24\\" width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><polygon points=\\"20.374 9 21 9.78 12 17 3 9.78 3.626 9 12 15.718\\"></polygon></svg></button><div id=\\"abcdef\\" class=\\"css-1cn06ep-Wrapper e2613d74\\"><ul class=\\"css-5ovq0u-List e2613d73\\"><li><a class=\\"css-1tupqjh-Link e2613d71\\" href=\\"/projects/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/resource/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-1juodu2-LinkIcon e2613d70 svg-icon--eyeOpen32\\" width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 32 32\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M16.0027,12.509 C14.3487,12.509 13.0027,13.855 13.0027,15.509 C13.0027,17.163 14.3487,18.509 16.0027,18.509 C17.6567,18.509 19.0027,17.163 19.0027,15.509 C19.0027,13.855 17.6567,12.509 16.0027,12.509 M16.0027,19.509 C13.7967,19.509 12.0027,17.715 12.0027,15.509 C12.0027,13.303 13.7967,11.509 16.0027,11.509 C18.2087,11.509 20.0027,13.303 20.0027,15.509 C20.0027,17.715 18.2087,19.509 16.0027,19.509 M5.242,15.508 C6.262,16.807 10.683,22.015 16.003,22.015 C21.322,22.015 25.743,16.808 26.763,15.508 C25.743,14.208 21.322,9 16.003,9 C10.675,9 6.261,14.208 5.242,15.508 M16.003,23.015 C9.437,23.015 4.418,16.093 4.207,15.799 L4,15.508 L4.207,15.217 C4.418,14.923 9.437,8 16.003,8 C22.568,8 27.588,14.923 27.798,15.217 L28.006,15.508 L27.798,15.799 C27.588,16.093 22.568,23.015 16.003,23.015\\"></path></svg><span class=\\"css-1avywkn-LinkText e2613d72\\">actions.view</span></a></li></ul></div></div></div><div class=\\"css-18isdm2-UtilityRow e1yvm5521\\"><div class=\\"utility-button-group utility-button-group--inline\\"><a class=\\"utility-button\\" href=\\"/projects/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/resource/slug-AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg utility-button__icon svg-icon--eyeOpen32\\" width=\\"26\\" height=\\"26\\" viewBox=\\"0 0 32 32\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M16.0027,12.509 C14.3487,12.509 13.0027,13.855 13.0027,15.509 C13.0027,17.163 14.3487,18.509 16.0027,18.509 C17.6567,18.509 19.0027,17.163 19.0027,15.509 C19.0027,13.855 17.6567,12.509 16.0027,12.509 M16.0027,19.509 C13.7967,19.509 12.0027,17.715 12.0027,15.509 C12.0027,13.303 13.7967,11.509 16.0027,11.509 C18.2087,11.509 20.0027,13.303 20.0027,15.509 C20.0027,17.715 18.2087,19.509 16.0027,19.509 M5.242,15.508 C6.262,16.807 10.683,22.015 16.003,22.015 C21.322,22.015 25.743,16.808 26.763,15.508 C25.743,14.208 21.322,9 16.003,9 C10.675,9 6.261,14.208 5.242,15.508 M16.003,23.015 C9.437,23.015 4.418,16.093 4.207,15.799 L4,15.508 L4.207,15.217 C4.418,14.923 9.437,8 16.003,8 C22.568,8 27.588,14.923 27.798,15.217 L28.006,15.508 L27.798,15.799 C27.588,16.093 22.568,23.015 16.003,23.015\\"></path></svg><span class=\\"utility-button__text\\">actions.view</span></a></div></div><nav class=\\"ecdc3150 css-8n39ik-Nav-SecondaryNav e1vnhept5\\"><button class=\\"css-iszmcy-Disclosure e1vnhept3\\"><div class=\\"css-1wmeakp-Active e1vnhept4\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg css-9dlcye-Icon e1vnhept2 svg-icon--disclosureDown16\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><polygon points=\\"8 9.702 13.356 5 13.999 5.733 8 10.999 2 5.733 2.643 5\\"></polygon></svg></div></button><ul class=\\"css-62htyi-List e1vnhept1\\"><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/properties\\">titles.properties</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/variants\\">titles.variants</a></li><li><a class=\\"css-slu7jg-Link e1vnhept0\\" href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/metadata\\">titles.metadata</a></li></ul></nav></div></header><div class=\\"backend-panel\\"><div class=\\"container\\"><aside class=\\"aside\\"><nav class=\\"secondary-nav panel-nav\\" aria-label=\\"resources.settings\\"><ul><li><a href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/properties\\">titles.properties</a></li><li><a href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/variants\\">titles.variants</a></li><li><a href=\\"/backend/projects/resource/AAAAAAAA-BBBB-CCCC-DDDD-EEEEEEEEEEEE/metadata\\">titles.metadata</a></li></ul></nav></aside><div class=\\"panel\\"><div></div></div></div></div>"`;
